web/src/components/storage/ProposalTransactionalInfo.jsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { Alert } from "@patternfly/react-core";
+import { sprintf } from "sprintf-js";
+
+import { _ } from "~/i18n";
+import { If, Section } from "~/components/core";
+import { isTransactionalSystem } from "~/components/storage/utils";
+import { useProduct } from "~/context/product";
+
+/**
+ * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
```

Comment:
Note for future change:  this is actually "not working". If you add the `@ts-check` _comment directive_ to the file and then run the `npm run check-types` task it will complain because

```
src/components/storage/ProposalTransactionalInfo.jsx:34:42 - error TS2694: Namespace '"/src/client/storage"' has no exported member 'ProposalManager'.

34  * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
                            
```

And it is right. The `ProposalManager` class is actually not exported. I think there will be changes in the clients soon, not sure. But in any case, I'll move all the type definitions to the top of the file, avoiding to have them _name spaced_ unless it make sense. @imobachgs, friendly ping.